NebulaWorld/Logistics/ILSShipManager.cs

```diff
@@ -209,13 +218,64 @@ public static void AddTakeItem(ILSShipItems packet)
                 }
                 else
                 {
                     Log.Info($"Calling TakeItem() with item {packet.itemId} and amount {packet.itemCount}");
                     int itemId = packet.itemId;
                     int itemCount = packet.itemCount;
                     stationComponent.TakeItem(ref itemId, ref itemCount);
+                    if(stationComponent.workShipDatas[packet.origShipIndex].otherGId != 0 && stationComponent.workShipDatas[packet.origShipIndex].otherGId < GameMain.data.galacticTransport.stationPool.Length)
+                    {
+                        StationComponent otherStationComponent = GameMain.data.galacticTransport.stationPool[stationComponent.workShipDatas[packet.origShipIndex].otherGId];
+                        if (otherStationComponent != null && otherStationComponent.storage != null)
+                        {
+                            for (int i = 0; i < otherStationComponent.storage.Length; i++)
+                            {
+                                if (otherStationComponent.storage[i].itemId == packet.itemId)
+                                {
+                                    otherStationComponent.storage[i].remoteOrder += packet.itemCount;
+                                    RefreshValuesUI(otherStationComponent, i);
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                    for (int i = 0; i < stationComponent.storage.Length; i++)
+                    {
+                        if (stationComponent.storage[i].itemId == packet.itemId)
+                        {
+                            if (stationComponent.workShipDatas[packet.origShipIndex].direction == 0)
+                            {
+                                stationComponent.storage[i].remoteOrder += packet.itemCount;
+                                RefreshValuesUI(stationComponent, i);
+                            }
+                            break;
+                        }
+                    }
+                }
+                Debug.Log(GameMain.galaxy.PlanetById(stationComponent.planetId).displayName + ": dir: " + stationComponent.workShipDatas[packet.origShipIndex].direction + " carry " + stationComponent.workShipDatas[packet.origShipIndex].itemId + "(" + stationComponent.workShipDatas[packet.origShipIndex].itemCount + ")");
+            }
+        }
+
+        private static void RefreshValuesUI(StationComponent stationComponent, int storageIndex)
+        {
+            UIStationWindow stationWindow = UIRoot.instance.uiGame.stationWindow;
+            if (stationWindow != null && (int)AccessTools.Field(typeof(UIStationWindow), "_stationId")?.GetValue(stationWindow) == stationComponent.id)
+            {
+                UIStationStorage[] stationStorageUI = (UIStationStorage[])AccessTools.Field(typeof(UIStationWindow), "storageUIs").GetValue(stationWindow);
```

Comment:
please cache these 3 AccessTools into FieldRef delegates. From the looks of it, these are called often, and they create allocations - possibly introducing more points for host/client to deync.

Reply:
included in the next commit.